Test TestCodexStaticModelsIncludeGPT55 fails on free tier after GPT-5.5 removal

## Bug

The test `TestCodexStaticModelsIncludeGPT55` fails on the `free` subtest because GPT-5.5 was removed from `codex-free` in `7b89583c` but the test still expects it there.

## Reproduction

```bash
go test -v -run TestCodexStaticModelsIncludeGPT55 ./internal/registry/
```

Output:
```
=== RUN   TestCodexStaticModelsIncludeGPT55/free
    model_definitions_test.go:17: expected codex free tier to include gpt-5.5
--- FAIL: TestCodexStaticModelsIncludeGPT55 (0.00s)
    --- FAIL: TestCodexStaticModelsIncludeGPT55/free (0.00s)
    --- PASS: TestCodexStaticModelsIncludeGPT55/team (0.00s)
    --- PASS: TestCodexStaticModelsIncludeGPT55/plus (0.00s)
    --- PASS: TestCodexStaticModelsIncludeGPT55/pro (0.00s)
```

## Root Cause

Commit `7b89583c` (`chore(models): remove GPT-5.5 model entry from registry JSON`) correctly removed GPT-5.5 from the `codex-free` section of `models.json` since free-tier accounts cannot access GPT-5.5. However, the test in `internal/registry/model_definitions_test.go` was not updated and still includes `"free": GetCodexFreeModels()` in its tier map.

## Suggested Fix

1. Remove `"free"` from the tier map in `TestCodexStaticModelsIncludeGPT55`
2. Add a new test `TestCodexFreeModelsExcludeGPT55` to explicitly verify the free tier exclusion

I have a PR ready with this fix.

--- a/internal/registry/model_definitions_test.go
+++ b/internal/registry/model_definitions_test.go
@@ -2,9 +2,15 @@ package registry
 
 import "testing"
 
+func TestCodexFreeModelsExcludeGPT55(t *testing.T) {
+	model := findModelInfo(GetCodexFreeModels(), "gpt-5.5")
+	if model != nil {
+		t.Fatal("expected codex free tier to NOT include gpt-5.5")
+	}
+}
+
 func TestCodexStaticModelsIncludeGPT55(t *testing.T) {
 	tierModels := map[string][]*ModelInfo{
-		"free": GetCodexFreeModels(),
 		"team": GetCodexTeamModels(),
 		"plus": GetCodexPlusModels(),
 		"pro":  GetCodexProModels(),

